src/network/peers.rs

```diff
@@ -598,29 +598,23 @@ impl<N: Network, E: Environment> Peers<N, E> {
 
                         true
                     }
-                    Message::UnconfirmedBlock(_, _, ref mut data) => {
-                        let block = if let Data::Object(block) = data {
-                            block
-                        } else {
-                            panic!("Logic error: the block shouldn't have been serialized yet.");
-                        };
-
+                    Message::UnconfirmedBlock(block_height, block_hash, ref mut data) => {
```

Comment:
Same comment as in https://github.com/AleoHQ/snarkOS/pull/1394#discussion_r766332446

Please revert this until a more thorough assessment is made.

The implication is that a malicious block from modified code can admit this change and poison internal last_seen filters if it is incorrect.

A simple fix is to check that block_height == block.height() and block_hash == block.hash() here, however this results in redundant checks. I can take a look at this change more closely later on to ensure we have it enforced at the lowest level possible.

Reply:
It seems like these values are already being checked on receipt: https://github.com/AleoHQ/snarkOS/blob/95a156cda79b4ee411c237089ec3c1a1f9fde514/src/network/peers.rs#L1212

I haven't seen any other relevant flows for these values, so I'm assuming this change would be safe?

Reply:
I still don't believe this is safe, but to keep things moving, I will merge it in. The current assumptions are fine, future assumptions won't necessarily know this flaw.